packages/neo-one-smart-contract-compiler/src/compile/syscalls.ts

```diff
@@ -977,6 +977,15 @@ export const SYSCALLS = {
     name: 'Neo.Storage.GetContext',
     returnType: StorageContextValue,
   }),
+  'Neo.Storage.GetReadOnlyContext': new SimpleSysCall({
+    name: 'Neo.Storage.GetReadOnlyContext',
+    returnType: StorageContextValue,
+  }),
+  'Neo.StorageContext.AsReadOnly': new SimpleSysCall({
+    name: 'Neo.StorageContext.AsReadOnly',
+    args: [new SysCallArgument('context', StorageContextValue)],
+    returnType: StorageContextValue,
+  }),
   'Neo.Storage.Get': new SimpleSysCall({
     name: 'Neo.Storage.Get',
     args: [new SysCallArgument('context', StorageContextValue), new SysCallArgument('key', StorageKey)],
```

Comment:
And then have this take in a union of `StorageContextValue` and `StorageContextReadonlyValue`

Reply:
Same for elsewhere that are read operations ('Neo.Storage.Find' might be the only other place)